packages/ui-keyring/src/Keyring.ts

```diff
@@ -237,13 +237,8 @@ export class Keyring extends Base implements KeyringStruct {
       }
     };
 
-    if (address.length === 42) {
-      this.accounts.add(this._store, address, json, 'ethereum');
-    } else {
-      const pair = this.keyring.addFromAddress(address, json.meta);
-
-      this.accounts.add(this._store, pair.address, json, pair.type);
-    }
+    const pair = this.keyring.addFromAddress(address, json.meta);
+    this.accounts.add(this._store, pair.address, json, address.length === 42? 'ethereum' : pair.type);
```

Comment:
This should not be needed after the adjustment in https://github.com/polkadot-js/extension/pull/572 (type is actually now returned)